### ISSUE TITLE
unable to download file in headless edge

### ISSUE THREAD

**Issue by prtk19** [MicrosoftEdge/EdgeWebDriver#43]
Hi, I am using edge 100.0.1185.39, edge driver 100.0.1185.39, selenium 4.1.0 and python 3.9
please find below code for launching edge in headless

```
def create_edge_driver(self, download_dir=""):
        selenium = BuiltIn().get_library_instance("SeleniumLibrary")
        download_directory = download_dir if download_dir != "" else os.path.join(os.path.expanduser("~"),'Desktop')
        options = webdriver.EdgeOptions()
        options.add_experimental_option("detach",True)      #prevents from closing edge browser
        options.add_experimental_option("prefs", {
            "download.default_directory": download_directory
        })
        options.add_argument("headless")
        options.add_argument("--no-sandbox")
        options.add_argument("--disable-single-click-autofill")
        options.add_argument("--disable-extensions")
        options.add_argument("--test-type")
        options.add_argument("--disable-site-isolation-trials")
        options.add_argument("--disable-web-security")
        options.add_argument("--disable-infobars")  # disable infobar
        options.add_argument("window-size=1920,1080")
        options.add_experimental_option("excludeSwitches", ['enable-automation']);
        options.add_argument("--disable-gpu")                
        options.add_argument("--disable-dev-shm-usage")      #overcome resource crunch issue
        options.add_argument('--ignore-certificate-errors')
        index = selenium.create_webdriver('Edge',
                                          options=options)
        browser = selenium.driver
        return browser
```
I am able to download file when I run test case in GUI mode, however same test case fails while running in headless mode.
Am I missing something? Please guide.

**Comment by bwalderman** [comment]
Hi @prtk19. Does the same issue occur in Chrome with ChromeDriver for GUI vs headless mode?

**Comment by prtk19** [comment]
Hi @bwalderman, I am using following code for downloading in chrome headless mode 

`def enable_download_in_headless_chrome(self, browser, download_dir=""):
        browser.command_executor._commands["send_command"] = (
            "POST", '/session/$sessionId/chromium/send_command')
        download_directory = download_dir if download_dir != "" else os.path.join(
                                                os.path.expanduser("~"), 'Desktop')
        params = {
            'cmd': 'Page.setDownloadBehavior',
            'params': {
                'behavior': 'allow',
                'downloadPath': download_directory
            }
        }
        browser.execute("send_command", params)`

and file do gets downloaded.
I also tried to use same code for edge(thinking it also uses chromium) headless but it was unable to download.

**Comment by bwalderman** [comment]
Thank you, that command should work for Edge as well since as you mentioned it uses Chromium. I have added this issue to our internal backlog for investigation.

**Comment by bwalderman** [comment]
@prtk19 can you share a minimal github repo that reproduces the issue? This will help us investigate and find the root cause more quickly.

**Comment by prtk19** [comment]
@bwalderman Sorry for late reply. I was out due to other commitments. 
I have connected with the my developer earlier, and the issue was with the file download window. So basically after clicking on file download option it opens a separate file download window. 
In headless mode, this window was opening for only micro seconds and as we had script to test this window, it was failing.
After this fix I retested the whole script and it worked.
Thanks again for taking it on priority.
